Tests/UI/BillingForm/Validator/PhoneNumberValidatorTests.swift

```diff
@@ -212,4 +214,49 @@ class PhoneNumberValidatorTests: XCTestCase {
         XCTAssertTrue(validator.isValid(text: testNumber))
         XCTAssertEqual(validator.formatForDisplay(text: testNumber), "+86 21 8051 0459")
     }
+    
+    func testSingletonBehaviour() {
+        let testCountryCode = "GB"
+        
+        // GIVEN I need the validator singleton, I will receive a valid singleton to use
+        weak var originalFormatter = PhoneNumberValidator.shared
+        originalFormatter?.countryCode = testCountryCode
+        XCTAssertEqual(originalFormatter?.countryCode, testCountryCode)
```

Comment:
I think this assertion is redundant and will always pass

Reply:
It prevents hidden passing for the final assertion.

If the object was not correctly deallocated and simply "appeared to not be available", the final part
```Swift
let newFormatter = PhoneNumberValidator.shared
XCTAssertNil(originalFormatter)
XCTAssertNotEqual(newFormatter.countryCode, testCountryCode)
``` 
may simply fail because the country code was never assigned correctly, whilst the object being identical to the first one.

So ensuring the property assignment behaves as expected, allows us to conclude in the final test that if that last assertion is failing, is only due to the original object being brought back to life instead of a new one (which never had the code assigned to begin with).